qiskit/aqua/utils/run_circuits.py

```diff
@@ -366,8 +358,8 @@ def run_on_backend(backend, qobj, backend_options=None,
                     config['noise_model'] = temp_noise_config
                 qobj.config = QasmQobjConfig.from_dict(config)
 
-            job = AerJob(backend, job_id, backend._run, qobj)
-            job.submit()
+            job = backend.run(qobj)
+            job.job_id = job_id
```

Comment:
```suggestion
            job.job_id = lambda: job_id
```

Reply:
Not setting the job_id anymore. I am letting the backend run method create it.